File: 2015/day02/erlang/test_solution.erl
-module(test_solution).
-export([run_tests/0]).

% Import the solution module functions
-import(solution, [calculate_wrapping_paper/3, parse_dimensions/1]).

% Simple test runner
run_test(Name, Expected, Actual) ->
    case Expected =:= Actual of
        true -> 
            io:format("✓ ~s~n", [Name]);
        false -> 
            io:format("✗ ~s - Expected: ~p, Got: ~p~n", [Name, Expected, Actual])
    end.

run_tests() ->
    io:format("Running Erlang Day 2 tests...~n"),
    
    % Test calculate_wrapping_paper with examples
    run_test("Example 1 (2x3x4 = 58)", 58, calculate_wrapping_paper(2, 3, 4)),
    run_test("Example 2 (1x1x10 = 43)", 43, calculate_wrapping_paper(1, 1, 10)),
    
    % Test edge cases
    run_test("Cube (2x2x2 = 28)", 28, calculate_wrapping_paper(2, 2, 2)),
    run_test("Unit cube (1x1x1 = 7)", 7, calculate_wrapping_paper(1, 1, 1)),
    
    % Test parse_dimensions
    run_test("Parse 2x3x4", {2, 3, 4}, parse_dimensions("2x3x4")),
    run_test("Parse 1x1x10", {1, 1, 10}, parse_dimensions("1x1x10")),
    run_test("Parse 100x200x300", {100, 200, 300}, parse_dimensions("100x200x300")),
    
    io:format("All tests completed!~n").

File: 2015/day02/lua/test_solution.lua
#!/usr/bin/env lua

-- Import solution functions
require("solution")

-- Simple test framework
local tests_passed = 0
local tests_total = 0

function run_test(name, expected, actual)
    tests_total = tests_total + 1
    if expected == actual then
        print("✓ " .. name)
        tests_passed = tests_passed + 1
    else
        print("✗ " .. name .. " - Expected: " .. tostring(expected) .. ", Got: " .. tostring(actual))
    end
end

function main()
    print("Running Lua Day 2 tests...")
    
    -- Test calculate_wrapping_paper with examples
    run_test("Example 1 (2x3x4 = 58)", 58, calculate_wrapping_paper(2, 3, 4))
    run_test("Example 2 (1x1x10 = 43)", 43, calculate_wrapping_paper(1, 1, 10))
    
    -- Test edge cases
    run_test("Cube (2x2x2 = 28)", 28, calculate_wrapping_paper(2, 2, 2))
    run_test("Unit cube (1x1x1 = 7)", 7, calculate_wrapping_paper(1, 1, 1))
    
    -- Test parse_dimensions
    local l1, w1, h1 = parse_dimensions("2x3x4")
    run_test("Parse 2x3x4 - length", 2, l1)
    run_test("Parse 2x3x4 - width", 3, w1)  
    run_test("Parse 2x3x4 - height", 4, h1)
    
    local l2, w2, h2 = parse_dimensions("1x1x10")
    run_test("Parse 1x1x10 - length", 1, l2)
    run_test("Parse 1x1x10 - width", 1, w2)
    run_test("Parse 1x1x10 - height", 10, h2)
    
    print("Tests completed: " .. tests_passed .. "/" .. tests_total .. " passed")
end

main()